docs/guide/service/nlb.md

```diff
@@ -1,74 +1,149 @@
-    For backwards compatibility, controller still supports the `nlb-ip` as the type annotation. For example, if you specify
-
-    ```
-    service.beta.kubernetes.io/aws-load-balancer-type: nlb-ip
-    ```
+## Prerequisites
+* AWS Load Balancer Controller >= v2.2.0
+* For Kubernetes Services resources of type `LoadBalancer`:
+    * Kubernetes >= v1.20 or
+    * Kubernetes >= v1.19.10 for 1.19 or
+    * Kubernetes >= v1.18.18 for 1.18 or
+    * EKS >= v1.16
+* For Kubernetes Services resources of type `NodePort`:
+    * Kubernetes >= v1.16
+* For `ip` target type:
+    * Pods have native AWS VPC networking configured, see [Amazon VPC CNI plugin](https://github.com/aws/amazon-vpc-cni-k8s)
 
-    the controller will provision NLB in IP mode. With this, the `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type` annotation gets ignored.
+## Configuration
 
-### Instance mode
-Similar to the IP mode, the instance mode is based on the annotation `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type` value `instance`. Here is a sample manifest snippet:
-!!!warning "NodePort allocation"
-    k8s version 1.22 and later support disabling NodePort allocation by setting the service field `spec.allocateLoadBalancerNodePorts` to `false`. If the NodePort is not allocated for a service port, the controller will fail to reconcile instance mode NLB.
-
-```yaml
-    metadata:
-      name: my-service
-      annotations:
-        service.beta.kubernetes.io/aws-load-balancer-type: "external"
-        service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "instance"
-```
+By default, Kubernetes Service resources of type `LoadBalancer` was reconciled by the Kubernetes controller built into the CloudProvider component of the kube-controller-manager or the cloud-controller-manager(a.k.a. the in-tree controller). 
+
+In order to let AWS Load Balancer Controller manage the reconciliation for Kubernetes Services resources of type `LoadBalancer`, we must offloading the reconciliation from in-tree controller to AWS Load Balancer Controller explicitly.
+
+
+=== "With LoadBalancerClass"
+    AWS Load Balancer Controller supports `LoadBalancerClass` feature since [:octicons-tag-24: v2.4.0](https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.4.0) release for Kubernetes v1.22+ clusters. 
+     
+    `LoadBalancerClass` feature provides a CloudProvider agnostic way of offloading the reconciliation for Kubernetes Services resources of type `LoadBalancer` to an external controller.
+    
+    When you specify the `spec.loadBalancerClass` to be `service.k8s.aws/nlb` on a Kubernetes Service resource of type `LoadBalancer`, the AWS Load Balancer Controller takes charge of reconciliation by provision an NLB.
+
+    !!! warning
+        - It's not recommended to modify or add the `spec.loadBalancerClass` on an existing Service resource. Instead, delete the existing Service resource and recreate a new one if a change is desired.
```

Comment:
`spec.loadBalancerClass` field is immutable. Users will not run into this situation.

Reply:
thanks for pointing this out